optimum/exporters/neuron/convert.py

```diff
@@ -549,32 +553,43 @@ def export_neuronx(
 
     compiler_args.extend(["--optlevel", optlevel])
 
-    # diffusers specific
-    compiler_args = add_stable_diffusion_compiler_args(config, compiler_args)
+    compiler_args = add_stable_diffusion_compiler_args(config, compiler_args)  # diffusers specific
 
     if config.dynamic_batch_size and not inline_weights_to_neff:
         logger.warning(
             "Dynamic batching is not yet compatible with the weights/neff non-inlined model. `inline_weights_to_neff` is set to True. If you still want to separate the neff and weights, please set `dynamic_batch_size=False`."
         )
         inline_weights_to_neff = True
 
-    neuron_model = neuronx.trace(
-        checked_model,
-        dummy_inputs_tuple,
-        compiler_args=compiler_args,
-        input_output_aliases=aliases,
-        inline_weights_to_neff=inline_weights_to_neff,
-        compiler_workdir=compiler_workdir,
-    )
-
-    if config.dynamic_batch_size is True:
-        neuron_model = neuronx.dynamic_batch(neuron_model)
-
-    # diffusers specific
-    improve_stable_diffusion_loading(config, neuron_model)
+    # Start trace
+    if tp_degree > 1:
+        # 1. use NxD to trace for parallel
+        neuron_model = neuronx_distributed.trace.parallel_model_trace(
```

Comment:
It is ok in a first step, but for LLama example they are not using this anymore, but instead the `ModelBuilder` class that wraps the model into `NxDModel` classes that contains several sub-models with different input shapes (bucketing).

Reply:
Is the use of bucketing mature and justified? I think we can start with `parallel_model_trace` anyway.

Reply:
It goes a bit beyond that, because prefill / decode already use two different input shapes, not even mentioning bucketing, and using the builder allows to share the same weights between all the alternate graphs.